pkg/streamingpromql/operators/functions/histogram_function.go

```diff
@@ -233,6 +233,7 @@ func (h *HistogramFunction) SeriesMetadata(ctx context.Context) ([]types.SeriesM
 	h.remainingGroups = make([]*bucketGroup, 0, len(groups))
 	for _, g := range groups {
 		seriesMetadata = append(seriesMetadata, types.SeriesMetadata{Labels: g.labels.DropMetricName()})
+		h.memoryConsumptionTracker.IncreaseMemoryConsumptionForLabels(g.labels)
```

Comment:
Shouldn't this be this?

```suggestion
		h.memoryConsumptionTracker.IncreaseMemoryConsumptionForLabels(g.labels.DropMetricName())
```

There are a bunch of places where we append a series and then update the memory consumption estimate. What if we introduced a wrapper `append` method that adds the labels to the slice and updates the memory consumption tracker? Then inconsistencies like this wouldn't be possible.